courses/serializers_test.py

```diff
@@ -77,45 +78,55 @@ def test_base_course_serializer():
     }
 
 
-@pytest.mark.parametrize("with_runs", [True, False])
-def test_serialize_course(mock_context, with_runs):
+@pytest.mark.parametrize("is_anonymous", [True, False])
+@pytest.mark.parametrize("show_all_runs", [True, False])
+def test_serialize_course(mock_context, is_anonymous, show_all_runs):
     """Test Course serialization"""
+    if is_anonymous:
+        mock_context["request"].user = AnonymousUser()
     user = mock_context["request"].user
+    if show_all_runs:
+        mock_context["show_all_runs"] = True
     course_run = CourseRunFactory.create(course__no_program=True, live=True)
     course = course_run.course
 
     # Create expired, enrollment_ended, future, and enrolled course runs
     CourseRunFactory.create(
-        course=course, end_date=datetime(2010, 1, 1, tzinfo=pytz.UTC)
+        course=course, end_date=datetime(2010, 1, 1, tzinfo=pytz.UTC), live=True
     )
     CourseRunFactory.create(
-        course=course, enrollment_end=datetime(2010, 1, 1, tzinfo=pytz.UTC)
+        course=course, enrollment_end=datetime(2010, 1, 1, tzinfo=pytz.UTC), live=True
     )
     CourseRunFactory.create(
-        course=course, enrollment_start=datetime(2119, 1, 1, tzinfo=pytz.UTC)
+        course=course, enrollment_start=datetime(2119, 1, 1, tzinfo=pytz.UTC), live=True
```

Comment:
Nit: I know this PR didn't introduce this specifically, but I'm wondering why these dates are hardcoded. Would be very easy to make them dynamic:

```python
from datetime import timedelta
now = now_in_utc()
CourseRunFactory.create(
        course=course, end_date=now - timedelta(1, "days"), live=True
)
```